### ISSUE TITLE
Entire test suite fails with Django 3.1b1 on Windows

### ISSUE THREAD

**Issue by smithdc1** [encode/django-rest-framework#7407]
When trying to look at the `url()` deprecation I installed Django 3.1b1. Every test fails with this version of Django.

Not sure if this should be raised here or as a ticket for Django. @carltongibson ?

I'm using Python 3.8.3 (latest release)

```
Python 3.8.3 (tags/v3.8.3:6f8c832, May 13 2020, 22:37:02) [MSC v.1924 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>>
```

```

_________________________ ERROR at setup of TestSlugRelatedField.test_overriding_get_queryset _________________________
..\venv\django-rest-framework\lib\site-packages\django\db\backends\utils.py:82: in _execute
    return self.cursor.execute(sql)
..\venv\django-rest-framework\lib\site-packages\django\db\backends\sqlite3\base.py:413: in execute
    return Database.Cursor.execute(self, query)
E   sqlite3.OperationalError: no such function: JSON_VALID

The above exception was the direct cause of the following exception:
..\venv\django-rest-framework\lib\site-packages\pytest_django\plugin.py:534: in _django_setup_unittest
    request.getfixturevalue("django_db_setup")
..\venv\django-rest-framework\lib\site-packages\pytest_django\fixtures.py:105: in django_db_setup
    db_cfg = setup_databases(
..\venv\django-rest-framework\lib\site-packages\django\test\utils.py:170: in setup_databases
    connection.creation.create_test_db(
..\venv\django-rest-framework\lib\site-packages\django\db\backends\base\creation.py:65: in create_test_db
    call_command(
..\venv\django-rest-framework\lib\site-packages\django\core\management\__init__.py:168: in call_command
    return command.execute(*args, **defaults)
..\venv\django-rest-framework\lib\site-packages\django\core\management\base.py:371: in execute
    output = self.handle(*args, **options)
..\venv\django-rest-framework\lib\site-packages\django\core\management\base.py:85: in wrapped
    res = handle_func(*args, **kwargs)
..\venv\django-rest-framework\lib\site-packages\django\core\management\commands\migrate.py:214: in handle
    self.sync_apps(connection, executor.loader.unmigrated_apps)
..\venv\django-rest-framework\lib\site-packages\django\core\management\commands\migrate.py:348: in sync_apps
    editor.create_model(model)
..\venv\django-rest-framework\lib\site-packages\django\db\backends\base\schema.py:324: in create_model
    self.execute(sql, params or None)
..\venv\django-rest-framework\lib\site-packages\django\db\backends\base\schema.py:142: in execute
    cursor.execute(sql, params)
..\venv\django-rest-framework\lib\site-packages\django\db\backends\utils.py:66: in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
..\venv\django-rest-framework\lib\site-packages\django\db\backends\utils.py:75: in _execute_with_wrappers
    return executor(sql, params, many, context)
..\venv\django-rest-framework\lib\site-packages\django\db\backends\utils.py:84: in _execute
    return self.cursor.execute(sql, params)
..\venv\django-rest-framework\lib\site-packages\django\db\utils.py:90: in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
..\venv\django-rest-framework\lib\site-packages\django\db\backends\utils.py:82: in _execute
    return self.cursor.execute(sql)
..\venv\django-rest-framework\lib\site-packages\django\db\backends\sqlite3\base.py:413: in execute
    return Database.Cursor.execute(self, query)
E   django.db.utils.OperationalError: no such function: JSON_VALID
```

**Comment by carltongibson** [comment]
This is a lack of JSON1 extension it looks like.

**Comment by carltongibson** [comment]
Install instructions are on the Django wiki 

https://code.djangoproject.com/wiki/JSON1Extension

(Looks like Python on Windows will ship with this enabled from 3.9)

**Comment by smithdc1** [comment]
Super, thanks. 

It's a bit well... ummm.. hard? I'm sure plenty of folks will be tripped up by this. But not obvious what we can do, is py 3.9 being released before Django 3.1? 

Another thought - could we add a check in Django that checks for the feature and can raise a better error message? Or not use it (it passes on 3.0?).

Or I could be worrying to much. 

Sorry not familiar with the details on this so lots of questions rather than something actually helpful.

**Comment by carltongibson** [comment]
Yes, it's not ideal. But Windows doesn't have an easier way of replacing shipped libraries. (TBH folks will struggle setting DYLib paths too, but other versions seem to already have this enabled, on the whole.)

Possibly we could have a system check, yes, but a lot of folks aren't using SQLite, and do you even hit this unless you're using JSONField? And so on... 

You could suggest one. Not sure what @felixxm would say... I seem to remember we batted the ball around over this already, but perhaps only between ourselves... 🤔

**Comment by felixxm** [comment]
> It's a bit well... ummm.. hard? I'm sure plenty of folks will be tripped up by this. But not obvious what we can do, is py 3.9 being released before Django 3.1?

No.

> Another thought - could we add a check in Django that checks for the feature and can raise a better error message? Or not use it (it passes on 3.0?).

We have a system check for this ([`fields.E180`](https://docs.djangoproject.com/en/3.1/ref/checks/#model-fields)).

> I seem to remember we batted the ball around over this already, but perhaps only between ourselves... thinking

We've tested the entire Django's test suite on Windows with SQLite without JSON1 extension and everything worked properly :thinking: 

I can check DRF tests with SQLite without JSON1 extension later (today or tomorrow).

**Comment by felixxm** [comment]
Fix is ready, see #7409.
